app/controllers/oaisys/pmh_controller.rb

```diff
@@ -2,25 +2,84 @@
 
 module Oaisys
   class PMHController < ApplicationController
+    EXPECTED_ARGUMENTS = {
+        identify: {
+            required: %w(),
+            optional: %w(),
+            exclusive: %w()
+        },
+        list_sets: {
+            required: %w(),
+            optional: %w(),
+            exclusive: %w(resumptionToken)
+        },
+        list_metadata_formats: {
+            required: %w(),
+            optional: %w(identifier),
+            exclusive: %w()
+        },
+        list_records: {
+            required: %w(metadataPrefix),
+            optional: %w(from until set),
+            exclusive: %w(resumptionToken)
+        },
+        get_record: {
+            required: %w(identifier metadataPrefix),
+            optional: %w(),
+            exclusive: %w()
+        },
+        list_identifiers: {
+            required: %w(metadataPrefix),
+            optional: %w(from until set),
+            exclusive: %w(resumptionToken)
+        },
+    }.freeze
     def bad_verb
     end
 
     def identify
+      expected_ags(__method__)
     end
 
     def list_sets
+      expected_ags(__method__)
     end
 
     def list_metadata_formats
+      expected_ags(__method__)
     end
 
     def list_records
+      expected_ags(__method__)
     end
 
     def get_record
+      expected_ags(__method__)
     end
 
     def list_identifiers
+      expected_ags(__method__)
+    end
+
+    private
+
+    def expected_ags (verb)
+      @verb = verb.to_s.camelize
+      @error_code = 'badArgument'
```

Comment:
Let's use a const for this instead

Reply:
I get a 'dynamic constant assignment' error if I define error_code as a constant here, maybe it should just be in the localization file?

Reply:
it's kind of moot now that I've realized we'll need to wrap this up in an exception class, but the constant probably needed to be set up at the class level. It's also probably better if this is a symbol, since there's no real reason for this to be mutable or have multiple instances or any of the things that strings give you.